source/common/router/rds_impl.cc

```diff
@@ -116,10 +117,23 @@ void RdsRouteConfigSubscription::onConfigUpdate(
     config_info_ = {new_hash, version_info};
     route_config_proto_ = route_config;
     stats_.config_reload_.inc();
-    ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
-              new_hash);
-    for (auto* provider : route_config_providers_) {
-      provider->onConfigUpdate();
+
+    if (!route_config_proto_.has_vhds()) {
+      ENVOY_LOG(debug, "rds: loading new configuration: config_name={} hash={}", route_config_name_,
+                new_hash);
+
+      last_update_details_ = std::make_unique<RdsConfigUpdateDetails>(route_config_proto_,
+                                                                      last_updated_, config_info_);
+      for (auto* provider : route_config_providers_) {
+        provider->onConfigUpdate();
+      }
+      vhds_subscription_.release();
+    } else {
+      auto s = new VhdsSubscription(route_config_proto_, factory_context_, stat_prefix_,
```

Comment:
Can you create the `std::make_unique` here? The idea is that generally in Envoy, we prefer to have smart pointers everywhere and it makes it easier to reason about cleanup and leaks if we don't need to think about whether code between this line and when we assign to the smart pointer might throw an exception etc.

Reply:
fixed